File: src/net/azib/photos/XMLListener.java
package net.azib.photos;

public interface XMLListener<T> {
  class StopParse extends Exception {}

  T getResult();

  void rootElement(String name);

  void rootElementEnd(String name);

  void value(String path, String value) throws StopParse;

  void start(String path) throws StopParse;

  void end(String path) throws StopParse;
}

File: src/net/azib/photos/XMLListener.kt
package net.azib.photos

interface XMLListener<out T> {
  class StopParse : Exception()

  val result: T

  fun rootElement(name: String) {}

  fun rootElementEnd(name: String) {}

  @Throws(StopParse::class)
  fun value(path: String, value: String)

  @Throws(StopParse::class)
  fun start(path: String) {}

  @Throws(StopParse::class)
  fun end(path: String) {}
}
